src/org/datafoodconsortium/connector/codegen/ruby/static/lib/datafoodconsortium/connector/importer.rb

```diff
@@ -0,0 +1,124 @@
+    class Importer
+      def self.type_map
+        unless @type_map
+          DataFoodConsortium::Connector.semantic_types.each do |type|
+            register_type(type)
+          end
+        end
+
+        @type_map
+      end
+
+      def self.register_type(clazz)
+        @type_map ||= {}
+        type_map[clazz::SEMANTIC_TYPE] = clazz
+      end
+
+      def self.prefixed_name(uri)
+        # When we skip backwards compatibility, we can just do this:
+        #
+        #     key = RDF::URI.new(uri).pname(prefixes: Context::VERSION_1_8)
+        #
+        # But for now we do it manually.
+        uri.gsub(
+          "https://github.com/datafoodconsortium/ontology/releases/latest/download/DFC_BusinessOntology.owl#",
+          "dfc-b:"
+        ).gsub(
+          # legacy URI
+          "http://static.datafoodconsortium.org/ontologies/DFC_BusinessOntology.owl#",
+          "dfc-b:"
+        )
+      end
+
+      def import(json_string_or_io)
+        @subjects = {}
+
+        graph = parse_rdf(json_string_or_io)
+        build_subjects(graph)
+        apply_statements(graph)
+
+        if @subjects.size > 1
+          @subjects.values
+        else
+          @subjects.values.first
+        end
+      end
+
+      private
+
+      # The `io` parameter can be a String or an IO instance.
+      def parse_rdf(io)
+        io = StringIO.new(io) if io.is_a?(String)
+        RDF::Graph.new << JSON::LD::API.toRdf(io)
+      end
+
+      def build_subjects(graph)
+        graph.query({ predicate: RDF.type }).each do |statement|
+          @subjects[statement.subject] = build_subject(statement)
+        end
+      end
+
+      def build_subject(type_statement)
+        # Not all subjects have an id, some are anonymous.
+        id = if type_statement.subject.respond_to?(:value)
+               type_statement.subject.value
+             end
+        type = type_statement.object.value
+        key = self.class.prefixed_name(type)
+        clazz = self.class.type_map[key]
+
+        clazz.new(*[id].compact)
+      end
+
+      def apply_statements(statements)
+        statements.each do |statement|
+          apply_statement(statement)
+        end
+      end
+
+      def apply_statement(statement)
+        subject = subject_of(statement)
+        property_uri = statement.predicate.value
+        value = resolve_object(statement.object)
+
+        property_id = self.class.prefixed_name(property_uri)
+
+        return unless subject.hasSemanticProperty?(property_id)
+
+        property = subject.semanticProperty(property_id)
+
+        if property.value.is_a?(Enumerable)
+          property.value << value
+        else
+          property.value = value
+        end
+      end
+
+      def subject_of(statement)
+        @subjects[statement.subject]
+      end
+
+      def resolve_object(object)
+        @subjects[object] || skos_concept(object) || object.object
+      end
+
+      def skos_concept(object)
+        return unless object.uri?
```

Comment:
Sometimes the object given is a literal, but still refers to skos concept, ie: `dfc-pt:drink`

```suggestion
        return unless object.value.match(":") || object.uri?
```